views/admission.py

```diff
@@ -382,17 +397,26 @@ def validate_field(request, admission_id):
 
 
 def _get_states_choices(accepted_states, admission, request):
-    if not request.user.has_perm('continuing_education.validate_registration') \
-            and admission.state in [REGISTRATION_SUBMITTED, VALIDATED]:
-        return []
-    else:
-        return [] if admission and admission.is_draft() else accepted_states.get('choices', ())
+
+    can_change_state = request.user.has_perm('continuing_education.change_admission_state', admission)
+    cannot_validate_registration = not request.user.has_perm('continuing_education.validate_registration', admission)
+    admission_is_draft = admission and admission.is_draft()
+    registration_is_submitted = admission.is_registration_submitted()
+    registration_is_validated = admission.is_validated()
+
+    if can_change_state:
+        if admission_is_draft:
+            return []
+        elif cannot_validate_registration and registration_is_submitted or registration_is_validated:
```

Comment:
Je serais explicite dans la priorité des opérateurs booléens : 
`cannot_validate_registration and (registration_is_submitted or registration_is_validated)`